packages/core/src/rum/instrumentation/DdRumUserInteractionTracking.tsx

```diff
@@ -44,9 +45,30 @@ export class DdRumUserInteractionTracking {
                     DdRumUserInteractionTracking.eventsInterceptor.interceptOnPress(...args)
                     return originalOnPress(...args)
                 }
+                // we store the original onPress prop so we can keep memoization working
+                props.__DATADOG_INTERNAL_ORIGINAL_ON_PRESS__ = originalOnPress
             }
             return original(element, props, ...children)
         }
+
+        const originalMemo = React.memo;
+        React.memo = (component: any, propsAreEqual?: (prevProps: any, newProps: any) => boolean) => {
+            return originalMemo(component, (prev, next) => {
+                if (!next.onPress) {
+                    return !!propsAreEqual ? propsAreEqual(prev, next) : areObjectShallowEqual(prev, next);
+                }
+                // we replace "our" onPress from the props by the original for comparison
+                const {onPress: _prevOnPress, ...partialPrevProps  } = prev;
+                const prevProps = {...partialPrevProps, onPress: prev.__DATADOG_INTERNAL_ORIGINAL_ON_PRESS__}
+                
+                const {onPress: _nextOnPress, ...partialNextProps  } = next;
+                const nextProps = {...partialNextProps, onPress: next.__DATADOG_INTERNAL_ORIGINAL_ON_PRESS__}
+
+                // if no comparison function is provided we do shallow comparison
+                return !!propsAreEqual ? propsAreEqual(prevProps, nextProps) : areObjectShallowEqual(nextProps, prevProps);
```

Comment:
So in the end we just do a custom `memo` comparison function which constructs fake props just for the sake of the comparison? Do I get it properly?

Reply:
I am not sure what you mean by fake props, so let me give more details.
During the `createElement` phase, we modify the `onPress` prop and store the original under another name.
To make this operation "transparent" for the memo phase, we replace our modified `onPress` by the original before doing the comparison.

I have added a small drawing, feel free to call me if this is still not 100% clear.

![image](https://user-images.githubusercontent.com/8973379/167095475-bf42cede-26b4-448f-9e9e-f2d96558c56a.png)

Reply:
thanks, it is clear :) I just meant that `prevProps` and `nextProps` are not attached to any component and just created to do a comparison.